### PR TITLE
Add automatic Docker image build

### PR DESCRIPTION
**Pull request by Ftonans**
I think it would be convenient to have latest images build automatically (so people don't have to build them on their servers), except I think I also should add dev version of a container since releases are quite rare.

This code allows for automatic image build on release.

### CODE REVIEW

**Review by syeopite — COMMENTED** [review]
Thanks for the PR!

I'll prefer it if Priviblur can be published on Quay though similar to how other Privacy frontend projects are.

And yes a tag for the latest unstable commit from the master branch would be great to have

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_ROBOT_TOKEN }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v5
+        with:
+          context: docker
+          push: true
+          tags: syeopite/priviblur:latest-unstable
```
Comment (TheFrenchGhosty) [inline .github/workflows/docker.yml:34]: -dev or -master would be more obvious

Comment (TheFrenchGhosty) [reply .github/workflows/docker.yml]: @Ftonans I meant just "dev", not "latest-dev"

File: .github/workflows/release-docker.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+name: Create and publish a Docker image
+
+on:
+  push:
+    branches:
+      - 'master'
+    tags:
+      - 'v*'
```
Comment (syeopite) [inline .github/workflows/release-docker.yml:8]: Shouldn't this be?

```
on:
  release:
    types: [published]
```

File: .github/workflows/docker.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+name: Create and publish a Docker image
+
+on:
+  push:
+    branches:
+      - 'master'
+    paths-ignore:
+      - 'config.example.toml'
+      - '.gitignore'
+      - 'docker-compose.yml'
```
Comment (syeopite) [inline .github/workflows/docker.yml:10]: Why is the compose and example config file excluded?

Comment (syeopite) [reply .github/workflows/docker.yml:10]: I'll prefer it if you can remove those two files from being excluded.

Changes to the compose (which will probably be edited to use the image) and the example config is something I'll like to keep updated in the image

File: .github/workflows/docker.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_ROBOT_TOKEN }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v5
+        with:
+          context: docker
+          push: true
+          tags: syeopite/priviblur:latest-dev
```
Comment (TheFrenchGhosty) [inline .github/workflows/docker.yml:34]: @Ftonans I meant just "dev", not "latest-dev"

**Review by TheFrenchGhosty — COMMENTED** [review]
I'm not an "expert" of GitHub CI, but this PR seems okay to me, even if it's far from perfect:

- ARM isn't supported for example
- Images never expires
- ...

I never personally made images with releases so I don't really know if the way you did it is correct, but it should be fine.

The rest seems OK

**Comment by syeopite** [comment]
Following a discussion I had with @TheFrenchGhosty, I'm only going to be officially supporting the release-image.

The "official" unstable image that builds off master would instead be pussthecat.org's image here https://github.com/PussTheCat-org/docker-priviblur-quay

Please remove the CI for the unstable-image, and address the issues @TheFrenchGhosty mentioned for the release-image.

My apologies about the extra work.

**Pull request merged**